### PR TITLE
Pull up reusable metrics to top level interface.

### PR DESCRIPTION
**Pull request by sophokles73**
* add metric for reporting anonymous connections
* fix InfluxDB template for connection metrics
* use consistent approach for generating Graphite metrics names

Signed-off-by: Kai Hudalla <kai.hudalla@bosch-si.com>

### CODE REVIEW

File: site/content/api/Metrics.md [hunk 1]
```diff
@@ -48,9 +48,35 @@ extracting the following *tags* from metric names transmitted via the Graphite r
...
+| *meter.hono.commands.response.delivered.count*     | *host*, *tenant*, *protocol*     | Command responses delivered to applications. Total count since application startup. |
+| *meter.hono.commands.response.delivered.m1_rate*   | *host*, *tenant*, *protocol*     | Command responses delivered to applications. One minute, exponentially weighted, moving average. |
+| *meter.hono.commands.response.delivered.m5_rate*   | *host*, *tenant*, *protocol*     | Command responses delivered to applications. Five minute, exponentially weighted, moving average. |
+| *meter.hono.commands.response.delivered.m15_rate*  | *host*, *tenant*, *protocol*     | Command responses delivered to applications. Fifteen minute, exponentially weighted, moving average. |
+| *meter.hono.commands.response.delivered.mean_rate* | *host*, *tenant*, *protocol*     | Command responses delivered to applications. Mean rate of messages since the application start. |
+| *meter.hono.commands.ttd.expired.count*            | *host*, *tenant*, *protocol*     | Messages containing a TTD that expired with no pending command(s). Total count since application startup. |
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/content/api/Metrics.md]: not right now, but @kaniyan is already working on adding this to the MQTT adapter. And these metrics are definitely intended to be collected for all protocol adapters, right?

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java [hunk 2]
```diff
@@ -948,12 +948,12 @@ protected final void close(final MqttEndpoint endpoint, final Device authenticat
         onClose(endpoint);
         sendDisconnectedEvent(endpoint.clientIdentifier(), authenticatedDevice);
         if (authenticatedDevice == null) {
-            LOG.debug("connection to unauthenticated device [clientId: {}] closed", endpoint.clientIdentifier());
-            metrics.decrementUnauthenticatedMqttConnections();
+            LOG.debug("connection to anonymous device [clientId: {}] closed", endpoint.clientIdentifier());
+            metrics.decrementAnonymousConnections();
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java]: are you referring to the log statement?

File: site/content/api/Metrics.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pellmann) [reply site/content/api/Metrics.md]: Hmm - you could argue, that they all could be used. But for MQTT the ttd will never expire, so this specific metric will not be collected for MQTT.

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (pellmann) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java]: No, to the changed method/metric name from Unauthenticated to Anonymous.

File: site/content/api/Metrics.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaniyan) [reply site/content/api/Metrics.md]: As part of metrics for MQTT Adapter, I also pulled up the methods to the Metrics level except incrementNoCommandReceivedAndTTDExpired(). As Marc said, it is applicable only to HTTP Adapter.

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sophokles73) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java]: ok, I'll change it back to *unauthenticated*

File: site/content/api/Metrics.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pellmann) [reply site/content/api/Metrics.md]: Okay, make sense.

**Comment by sophokles73** [comment]
@pellmann  I have pushed the requested change ...

File: adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java [hunk 3]
```diff
@@ -830,22 +830,22 @@ public void testConnectionMetrics() {
...
 
         closeHandlerRef.get().handle(null);
 
-        verify(metrics).decrementMqttConnections("DEFAULT_TENANT");
+        verify(metrics).decrementConnections("DEFAULT_TENANT");
     }
 
     /**
-     * Verifies the connection metrics for unauthenticated connections.
+     * Verifies the connection metrics for anonymous connections.
```
Comment (pellmann) [inline adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java]: anonymous -> unauthenticated?

File: adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java [hunk 4]
```diff
@@ -830,22 +830,22 @@ public void testConnectionMetrics() {
...
      * <p>
      * This test should check if the metrics receive a call to increment and decrement when a connection is being
      * established and then closed.
      */
     @SuppressWarnings("unchecked")
     @Test
-    public void testUnauthenticatedConnectionMetrics() {
+    public void testAnonymousConnectionMetrics() {
```
Comment (pellmann) [inline adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java]: anonymous -> unauthenticated?

**Pull request merged**